examples-gallery/plot_crane_moving_a_load.py

```diff
@@ -0,0 +1,294 @@
+- q: angle of the rod [rad]
+- uxc: velocity of the mover in x-direction [m/s]
+- uxl: velocity of the load in x-direction [m/s] dependent speed
+- uyl: velocity of the load in y-direction [m/s] dependent speed
+- u: angular velocity of the rod [rad/s]
+
+**Specifieds**
+
+- F: force applied to the mover [N]
+
+"""
+import sympy.physics.mechanics as me
+from collections import OrderedDict
+import numpy as np
+import sympy as sm
+from scipy.interpolate import CubicSpline
+from opty.direct_collocation import Problem
+from opty.utils import parse_free
+import matplotlib.pyplot as plt
+import matplotlib.animation as animation
+from matplotlib import patches
+
+# %%
+# Set up Kane's equations of motion.
+# Just the standard way of doing it.
+
+N, A = sm.symbols('N A', cls = me.ReferenceFrame)
+t = me.dynamicsymbols._t
+O, P1, P2 = sm.symbols('O P1 P2', cls = me.Point)
+O.set_vel(N, 0)
+xc, xl, yl, q, uxc, uxl, uyl, u, F = me.dynamicsymbols('xc, xl, yl, q, uxc,' +
+    'uxl, uyl, uq, F')
+l, m1, m2, g = sm.symbols('l, m1, m2, g')
+
+A.orient_axis(N, q, N.z)
+A.set_ang_vel(N, u * N.z)
+
+P1.set_pos(O, xc * N.x)
+P1.set_vel(N, uxc * N.x)
+
+P2.set_pos(P1, -l * A.y)
+P2.v2pt_theory(P1, N, A)
+
+P1a = me.Particle('P1a', P1, m1)
+P2a = me.Particle('P2a', P2, m2)
+
+BODY = [P1a, P2a]
+
+FL = [(P1, F * N.x - m1*g*N.y), (P2, -m2*g*N.y)]
+kd = sm.Matrix([uxc - xc.diff(t), u - q.diff(t), uxl - xl.diff(t),
+    uyl - yl.diff(t)])
+
+config_constr = sm.Matrix([xl - xc - l * sm.sin(q), yl + l * sm.cos(q)])
+speed_constr = config_constr.diff(t)
+
+q_ind = [xc, q]
+q_dep = [xl, yl]
+u_ind = [uxc, u]
+u_dep = [uxl, uyl]
+
+KM = me.KanesMethod(N,
+    q_ind=q_ind,
+    q_dependent=q_dep,
+    u_ind=u_ind,
+    u_dependent=u_dep,
+    kd_eqs=kd,
+    configuration_constraints=config_constr,
+    velocity_constraints=speed_constr)
+
+(fr, frstar) = KM.kanes_equations(BODY, FL)
+EOM = kd.col_join(fr + frstar)
+EOM = EOM.col_join(config_constr)
+sm.pprint(EOM)
+# %%
+# Set up the optimization problem and solve it.
+# Note: While opty could calculate the reaction forces on, say, the mover,
+# (add them to the state variables), my trials showed that even with this
+# small example the calculation time increases enourmously.
+
+state_symbols = tuple((*q_ind, *q_dep, *u_ind, *u_dep))
+constant_symbols = (l, m1, m2, g)
+specified_symbols = (F,)
+h_opty = sm.symbols('h_opty')
+unknown_symbols = []
+methode = "backward euler"
+
+num_nodes = 250
+duration =(num_nodes - 1) * h_opty
+interval_value = h_opty
+
+# Specify the known system parameters.
+par_map = OrderedDict()
+par_map[l] = 5.0
+par_map[m1] = 1.0
+par_map[m2] = 10.0
+par_map[g] = 9.81
+
+t0, tf = 0.0, duration
+
+def obj(free):
+    """Minimize h_opty, the time interval between nodes."""
+    return free[-1]
+
+def obj_grad(free):
+    grad = np.zeros_like(free)
+    grad[-1] = 1.
+    return grad
+
+# start location and and final location of the load.
+anfang = 0.0
+ende = 15.0
```

Comment:
It is preferable if the documentation is all in English to be consistent.

Reply:
> It is preferable if the documentation is all in English to be consistent.

I did this because in my early python times, I used English words, which were standard Python terms - and then I wondered why my program did not run (like I called a variable 'range'. Will change it to English.

Reply:
That's a good method, but we should just make the docs consistent in language. Sphinx does have mechanisms to make multiple versions of the docs in different languages. I've never done that, but I'm sure it would be helpful to many learners.